polyfactory/factories/base.py

```diff
@@ -175,6 +175,20 @@ class BaseFactory(ABC, Generic[T]):
     """
     Flag indicating whether to use the default value on a specific field, if provided.
     """
+    __extra_providers__: dict[Any, Callable[[], Any]] | None = None
```

Comment:
Nit: I think this should be named `_extra_providers` since we generally allow users to override the dunder fields.